api/src/main/java/com/capitalone/dashboard/webhook/github/GitHubCommitV3.java

```diff
@@ -0,0 +1,318 @@
+import java.util.stream.Collectors;
+
+public class GitHubCommitV3 extends GitHubV3 {
+    private static final Log LOG = LogFactory.getLog(GitHubCommitV3.class);
+
+    private final CommitRepository commitRepository;
+    private final GitRequestRepository gitRequestRepository;
+    private final GitHubRepoRepository gitHubRepoRepository;
+
+    public GitHubCommitV3(CollectorService collectorService,
+                          RestClient restClient,
+                          CommitRepository commitRepository,
+                          GitRequestRepository gitRequestRepository,
+                          GitHubRepoRepository gitHubRepoRepository,
+                          ApiSettings apiSettings) {
+        super(collectorService, restClient, apiSettings);
+
+        this.commitRepository = commitRepository;
+        this.gitRequestRepository = gitRequestRepository;
+        this.gitHubRepoRepository = gitHubRepoRepository;
+    }
+
+    @Override
+    public QueryDslPredicateExecutor<GitHubRepo> getGitHubRepoRepository() { return this.gitHubRepoRepository; }
+
+    @Override
+    public String process(JSONObject jsonObject) throws MalformedURLException, HygieiaException, ParseException {
+        String result = "Commits Processed Successfully";
+
+        String branch = null;
+        String repoUrl = null;
+
+        Object commitsObj = jsonObject.get("commits");
+        if (!(commitsObj instanceof List)) {
+            result = "No Commits Data Found";
+            return result;
+        }
+
+        List<Map> commitsObjectList = (ArrayList<Map>) jsonObject.get("commits");
+
+        if (CollectionUtils.isEmpty(commitsObjectList)) {
+            return "Commits JSONArray Empty.";
+        }
+
+        Object repoObject = jsonObject.get("repository");
+        if (!(repoObject instanceof Map)) {
+            return "No Repository Found";
+        }
+
+        repoUrl = restClient.getString(repoObject, "url");
+        String ref = restClient.getString(jsonObject, "ref");
+        if (!StringUtils.isEmpty(ref)) {
+            branch = ref.replace("refs/heads/", "");
+        }
+
+        Object senderObj = jsonObject.get("sender");
+        String authorLDAPDN = restClient.getString(senderObj,"ldap_dn");
+
+        List<Commit> commitList = getCommits(commitsObjectList, repoUrl, branch, authorLDAPDN);
+
+        commitRepository.save(commitList);
+
+        return result;
+    }
+
+    protected List<Commit> getCommits(List<Map> commitsObjectList, String repoUrl,
+                                      String branch, String authorLDAPDN) throws MalformedURLException, HygieiaException, ParseException {
+        List<Commit> commitsList = new ArrayList<>();
+
+        GitHubParsed gitHubParsed = new GitHubParsed(repoUrl);
+        CollectorItem collectorItem = getCollectorItem(gitHubParsed.getUrl(), branch);
+        GitHubWebHookSettings gitHubWebHookSettings = parseAsGitHubWebHook(apiSettings.getGitHubWebHook());
```

Comment:
handle null checks for gitHubWebHookSettings there is a possibility parseAsGitHubWebHook() can return it as null

Reply:
Good Catch. I have put this check in, thanks.